src/test/java/uk/ac/cam/cl/dtg/isaac/api/InfoFacadeTest.java

```diff
@@ -0,0 +1,171 @@
+package uk.ac.cam.cl.dtg.isaac.api;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+import org.apache.commons.lang3.SystemUtils;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+import uk.ac.cam.cl.dtg.isaac.IsaacTest;
+import uk.ac.cam.cl.dtg.isaac.api.managers.GameManager;
+import uk.ac.cam.cl.dtg.isaac.api.services.GroupChangedService;
+import uk.ac.cam.cl.dtg.segue.api.InfoFacade;
+import uk.ac.cam.cl.dtg.segue.api.managers.UserAccountManager;
+import uk.ac.cam.cl.dtg.segue.configuration.SegueGuiceConfigurationModule;
+import uk.ac.cam.cl.dtg.segue.dao.content.ContentMapper;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+import javax.ws.rs.core.EntityTag;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.Map;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static org.easymock.EasyMock.anyObject;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.createMock;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.DEFAULT_LINUX_CONFIG_LOCATION;
+
+@PowerMockIgnore("javax.net.ssl.*")
+public class InfoFacadeTest extends IsaacTest {
+
+    public InfoFacade infoFacade;
+
+    public Request requestForCaching;
+
+    @ClassRule
+    public static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:12")
+            .withEnv("POSTGRES_HOST_AUTH_METHOD", "trust")
```

Comment:
Surely it would be better to set an actual password here, loaded from the config file? I'm not clear on why this has to be an `@ClassRule` and isn't just in the `@Before` though, because accessing the property would be tricky as is.

Reply:
`@Before` runs before every test case. I'm not super sure about the distinction between a `@ClassRule` or `@BeforeClass` but the docs prefer the former and I remember reading up about it quite a bit.

AFAIR I set this up so that the testcontainers are singletons so only ever spun up once in practice but I may be misremembering here. Anyway, I've been prioritising getting things to run and the nice engineering will come later.

Reply:
Singletons are now a thing.